### DISCUSSION TITLE
Force Smart mode not working on Edge 530

### DISCUSSION THREAD

**Discussion by bzx** [maca88/SmartBikeLights#15] [General]
Me again with a question ;) This is again for Edge 530.

I've got both lights in Smart mode now (both showing "S" on the data field), and I've checked the "force smart mode" in the config. Things work fine, except that when I press the physical button on the light, it changes to manual mode, and doesn't go back into smart mode no matter what. Even if I stop and discard the activity, the light is still in manual mode, and doesn't turn on when another activity is started – this happens also after restarting the Edge, and re-establishing the light network.

Please let me know if there's something wrong with my config, or this is a known issue.
`#4587520,196641#2,2|:1:0:1H[2|:1:63:1D=1#6291461,1409482753#2,2|:1:0:1H[2|:1:7:1D=1#3:Ion 200 RT|Off:0|Day Flash:63|Night Flash:62#4:Flare RT|Off:0|All Day Flash:8|Night Flash:63|Night Steady:5#0::#1:1#B3121#2#1#0#0`

Thank you!

**Comment by maca88** [comment]
> Things work fine, except that when I press the physical button on the light, it changes to manual mode

Do you have the latest version of the data field (0.13.0)? The "force smart mode" feature was added in the latest release and it will not work with a previous version of the data field. I tested on the simulator for Edge 530 and it works fine, the Smart mode remains after switching the light mode by using the physical button on the light. 

 >  Even if I stop and discard the activity, the light is still in manual mode, and doesn't turn on when another activity is started – this happens also after restarting the Edge, and re-establishing the light network.

On Edge 530, the control mode is persisted so it will not change until changed by using the CIQ data field menu.

> Please let me know if there's something wrong with my config, or this is a known issue.

The configuration is fine.

**Reply by bzx** [reply]
> Do you have the latest version of the data field (0.13.0)? The "force smart mode" feature was added in the latest release and it will not work with a previous version of the data field. I tested on the simulator for Edge 530 and it works fine, the Smart mode remains after switching the light mode by using the physical button on the light.

I installed the data field 3 days ago – it's the latest version.

**Reply by maca88** [reply]
That is strange, I tested again on my Edge 1000 and it works fine... also in the forums the guy that requested this feature reported that [it works](https://forums.garmin.com/developer/connect-iq/f/showcase/248492/smart-bike-lights-data-field-bike-lights-control-widget/1296469#1296469), so I am really out of ideas what it could be wrong. Could you please test only with one light paired to see if it has any effect?

**Reply by bzx** [reply]
I just removed both lights from my 530, paired the rear Flare RT only. Since I already had the config loaded, when I started the activity the light started flashing in Smart mode, but as soon as I changed the mode directly on the lamp, it switched to Manual mode and didn't go back to Smart.

**Reply by maca88** [reply]
That is really frustrating, I am not able to reproduce it no matter what I try... unfortunately until I figure how to reproduce it, I am not able to help you :/ I will let you know in case I find the issue.

**Reply by bzx** [reply]
No prob, I can live with it ;) I'll probably stick to a manual mode anyway. If you want me to test an update at some point, let me know.

**Reply by maca88** [reply]
I found the issue and resolved it in the latest version `0.13.1`, please update and let me know if it works, thanks.

**Reply by bzx** [reply]
Just updated and tested with the config from the first comment, and it's still not working as expected:
1. After starting the activity the lights turn on with correct default mode, the data field is showing it properly too. 
2. After I change mode using buttons on the lights, they are staying in that mode for about 2-3 seconds, and turn back on to the default mode (the data field is still able to show the changed mode for a second or so).
3. Same behaviour for both lights.

Thanks!

**Reply by maca88** [reply]
Thanks for the feedback.

> After I change mode using buttons on the lights, they are staying in that mode for about 2-3 seconds, and turn back on to the default mode

This is the expected behavior, the data field cannot disable the light buttons from working, so the light mode will still change after pressing the button, but then the data field will send a signal to change back to the mode calculated by the filters from the Lights Configurator.

**Reply by bzx** [reply]
Got it. I had a false impression that in the Smart mode I can change the light mode with the physical button, and the light would be still in Smart mode and would change mode according to the next trigger from the config. For instance, I start in Smart mode with Day Flash, but I change it manually to Night Flash, but the light goes off after I stop the activity on Garmin (as I configured it for Smart mode). I agree that this is quite complicated and probably doesn't make sense for most ;) 

Anyhow, I realised that Smart mode is not for me, it's way faster to change mode with your hand than to click through the Garmin menus on 530 (while it's still possible in Manual mode if I actually want to do it).

Thanks!
